mitosheet/mitosheet/step_performers/column_steps/change_column_dtype.py

```diff
@@ -193,9 +193,9 @@ def execute(cls, prev_state: State, params: Dict[str, Any]) -> Tuple[State, Opti
             post_state.dfs[sheet_index][column_header] = new_column
             pandas_processing_time = perf_counter() - pandas_start_time
 
-            # If we're changing between number columns, we keep the formatting on the column. Otherwise, we remove it
-            if not ((is_int_dtype(old_dtype) or is_float_dtype(old_dtype)) and (is_int_dtype(new_dtype) or is_float_dtype(new_dtype))):
-                post_state.column_format_types[sheet_index][column_id] = {'type': FORMAT_DEFAULT}
+            # If we're changing away from a number column, then we remove the formatting on the column if it exists
+            if not is_number_dtype(new_dtype) and column_id in post_state.df_formats[sheet_index]['columns']:
+                del post_state.df_formats[sheet_index]['columns'][column_id]
```

Comment:
We don't. Currently, if you have a number format applied to a number column, and change it with a formula to a different type, it might still try and format it. 

I am fine with this - it's just a bit of extra generated code for a super rare case.